derive_client/_bridge/standard_bridge.py

```diff
@@ -0,0 +1,316 @@
+    L1_STANDARD_BRIDGE_ABI_PATH,
+    L2_CROSS_DOMAIN_MESSENGER_ABI_PATH,
+    L2_CROSS_DOMAIN_MESSENGER_PROXY,
+    L2_STANDARD_BRIDGE_ABI_PATH,
+    L2_STANDARD_BRIDGE_PROXY,
+    MSG_GAS_LIMIT,
+    RESOLVED_DELEGATE_PROXY,
+)
+from derive_client.data_types import (
+    Address,
+    BridgeTxDetails,
+    BridgeTxResult,
+    BridgeType,
+    ChainID,
+    Currency,
+    PreparedBridgeTx,
+    TxResult,
+)
+from derive_client.exceptions import BridgeEventParseError, PartialBridgeResult, StandardBridgeRelayFailed
+from derive_client.utils.w3 import to_base_units
+
+from .w3 import (
+    build_standard_transaction,
+    encode_abi,
+    get_contract,
+    get_w3_connections,
+    make_filter_params,
+    send_tx,
+    sign_tx,
+    wait_for_bridge_event,
+    wait_for_tx_finality,
+)
+
+
+def _load_l1_contract(w3: AsyncWeb3) -> AsyncContract:
+    address = L1_CHUG_SPLASH_PROXY
+    abi = json.loads(L1_STANDARD_BRIDGE_ABI_PATH.read_text())
+    return get_contract(w3=w3, address=address, abi=abi)
+
+
+def _load_l2_contract(w3: AsyncWeb3) -> AsyncContract:
+    address = L2_STANDARD_BRIDGE_PROXY
+    abi = json.loads(L2_STANDARD_BRIDGE_ABI_PATH.read_text())
+    return get_contract(w3=w3, address=address, abi=abi)
+
+
+def _load_l2_contracts(w3s: dict[ChainID, AsyncWeb3]) -> dict[ChainID, AsyncContract]:
+    return {chain_id: _load_l2_contract(w3) for chain_id, w3 in w3s.items() if chain_id is not ChainID.ETH}
+
+
+def _load_l1_cross_domain_messenger_proxy(w3: AsyncWeb3) -> AsyncContract:
+    address = RESOLVED_DELEGATE_PROXY
+    abi = json.loads(L1_CROSS_DOMAIN_MESSENGER_ABI_PATH.read_text())
+    return get_contract(w3=w3, address=address, abi=abi)
+
+
+def _load_l2_cross_domain_messenger_proxy(w3: AsyncWeb3) -> AsyncContract:
+    address = L2_CROSS_DOMAIN_MESSENGER_PROXY
+    abi = json.loads(L2_CROSS_DOMAIN_MESSENGER_ABI_PATH.read_text())
+    return get_contract(w3=w3, address=address, abi=abi)
+
+
+class StandardBridge:
+
+    def __init__(self, account: Account, logger: Logger):
+
+        self.account = account
+        self.logger = logger
+        self.w3s = get_w3_connections(logger=logger)
+        self.l1_contract = _load_l1_contract(self.w3s[ChainID.ETH])
+        self.l2_contracts = _load_l2_contracts(self.w3s)
+        self.l1_messenger_proxy = _load_l1_cross_domain_messenger_proxy(self.w3s[ChainID.ETH])
+        self.l2_messenger_proxy = _load_l2_cross_domain_messenger_proxy(self.w3s[ChainID.DERIVE])
+
+    @future_safe
+    async def prepare_eth_tx(
+        self,
+        human_amount: float,
+        to: Address,
+        source_chain: ChainID,
+        target_chain: ChainID,
+    ) -> IOResult[PreparedBridgeTx, Exception]:
+
+        currency = Currency.ETH
+
+        if source_chain is not ChainID.ETH or target_chain is not ChainID.DERIVE or to != self.account.address:
+            raise NotImplementedError("Only ETH transfers from Ethereum to Derive EOA are currently supported.")
+
```

Comment:
depositing ETH is different from erc20s. Just like the contract has different methods for it, so should we, as it provides the clearest interface. At the moment we only support gas funding from L1 to Derive L2 EOA.